Modules/xSharePoint/DSCResources/MSFT_xSPAppCatalog/MSFT_xSPAppCatalog.psm1

```diff
@@ -0,0 +1,79 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)] [System.String] $AppCatalogUrl ,
+        [parameter(Mandatory = $true)]  [System.String] $WebApp, 
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+
+    )
+
+    Write-Verbose -Message "Checking app urls settings"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+         $wa = Get-SPWebApplication $param.WebApp
+        $feature = $wa.Features | ?{ $_.ID -eq [Guid]::Parse("f8bea737-255e-4758-ab82-e34bb46f5828")}
+        if($feature -eq $null ){
+            throw "not an app catalog"
```

Comment:
Instead of throwing an exception here can we change it to return $null then check for $null in the test function?